### ISSUE TITLE
error_reporting

### ISSUE THREAD

**Issue by nika-begiashvili** [bcosca/fatfree#957]
could you please explain why there's  
`error_reporting((E_ALL|E_STRICT)&~(E_NOTICE|E_USER_NOTICE));`
inside constructor ?
it swallows undefined variable and similar errors

**Comment by bcosca** [comment]
A framework is not supposed to tolerate lazy and carefree coders.

**Comment by nika-begiashvili** [comment]
I don't understand your answer, why error reporting is set at all ? why override php.ini settings? and why is silencing some errors good idea ?

**Comment by bcosca** [comment]
The default php.ini settings are too permissive. Only notices are silenced. They are just informative. The rest should block operation.

**Comment by pgrzego** [comment]
I understand the concept of being able to pass all the warnings/errors through one ONERROR handler. But I think it would be a good idea to keep the info about the severity of the problem, so if I decide to overwrite F3 default handler, I will be able to react differently depending on whether it is a warning or an error.

Therefore my suggestion would be to change:

```php
set_error_handler(
	function($code,$text) use($fw) {
		if ($code & error_reporting())
			$fw->error(500,$text);
	}
);
```
into
```php
set_error_handler(
	function($code,$text) use($fw) {
		if ($code & error_reporting())
			$fw->error(500,$text,$code);
	}
);
```
Which I know will require refactoring of the `error` method, but imho it's worth it. Then, if `$code` will be populated, I will be able to write a custom ONERROR function:

```php
$f3->set("ONERROR", function($f3, $params, $code) {
  if ( $code == E_WARNING ) {
    // Don't stop at warnings:
    $f3->set("HALT", false);
    return true;
  } else {
    // Process more serious problems:
    echo "Error: ".$f3->get('ERROR.text')."\n";
    $f3->set("HALT", true);
    return false;
  }
});
```

**Comment by pgrzego** [comment]
> The rest should block operation and force you to fix your coding errors.

In general I agree with you. But I have a problem which is actually tricky to solve. I have a log file which is used by a F3 Log class. However it is used both by cron file which runs as root and by a regular front controller which runs as a regular user. Now if a cron will create the file first then the front controller gets a permission denied warning.

Now I can't override what Log class does. And I won't be checking for file permissions every time I want to log something. Maybe this problem is more for the F3 to solve? :smiling_imp:

**Comment by bcosca** [comment]
The addition of `ERROR.level` is a good idea. We can put this to rest now.
